x/superfluid/types/msgs.go

```diff
@@ -392,3 +393,31 @@ func (msg MsgAddToConcentratedLiquiditySuperfluidPosition) GetSigners() []sdk.Ac
 	}
 	return []sdk.AccAddress{sender}
 }
+
+var _ sdk.Msg = &MsgUnbondConvertAndStake{}
+
+func (msg MsgUnbondConvertAndStake) Route() string { return RouterKey }
+func (msg MsgUnbondConvertAndStake) Type() string {
+	return TypeMsgUnbondConvertAndStake
+}
+
+func (msg MsgUnbondConvertAndStake) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return fmt.Errorf("Invalid sender address (%s)", err)
+	}
+
+	return nil
+}
```

Comment:
We should be doing more formatting validation right?
e.g:
* val_addr either empty or valoper
* slippage bound non-negative

Reply:
This still needs to be addressed, along with respective validate basic test

Reply:
valoper still needs to be validated right?

Reply:
Right, good catch. Also, I don't think we need a val address validation here since we also do val address validation within the logic (if used)